classes/app.rb

```diff
@@ -47,7 +47,7 @@ def create_teacher
     age = gets.chomp.to_i
     print "Teacher's specialization: "
     specialization = gets.chomp
-    teacher = Teacher.new(age, specialization, name, parent_permission: true)
+    teacher = Teacher.new(age, specialization, name)
```

Comment:
- [ ] Hello, am going to leave comments for all changes that are required in the app.rb file because I can't find a suitable place to leave the comments

- 🎓👨‍🏫👩‍🏫 To follow the single responsibility principle effectively, the calls to create instances of the Teacher and Student classes, which both inherit from the Person class, should be consolidated within the create_person method. 🧐 This method has the exclusive responsibility of crafting either a teacher or a student object, making our code more organized and adherent to the principle. 🛠️ It's all about keeping things neat and efficient! 💡😊
- 📜🚀 The project requirements designated the main.rb file with the primary role of initiating the program and prompting the user for their initial choices or the option to quit. However, I've noticed that in your implementation, the user input requests and validation checks are placed in the app.rb file. To maintain a clear adherence to the single responsibility principle, it's advisable to centralize these user interactions within the main.rb file. 🔄 This separation of responsibilities ensures that each part of your program has a well-defined role, promoting better code organization and maintainability. 🧐✅
- 📜🚀 The project requirements designated the main.rb file with the primary role of initiating the program and prompting the user for their initial choices or the option to quit. However, I've noticed that in your implementation, the list of options are placed in the app.rb file. To maintain a clear adherence to the single responsibility principle, it's advisable to centralize these user interactions within the main.rb file. 🔄 This separation of responsibilities ensures that each part of your program has a well-defined role, promoting better code organization and maintainability. 🧐✅